engine/runtime/src/main/java/org/enso/interpreter/runtime/library/dispatch/TypeOfNode.java

```diff
@@ -75,6 +75,11 @@ Object doWarning(WithWarnings value, @Cached TypeOfNode withoutWarning) {
     return withoutWarning.execute(value.getValue());
   }
 
+  @Specialization
+  Object doEnsoMultiValue(EnsoMultiValue value) {
+    return value.allTypes();
```

Comment:
Hello Dmitry,
thanks for trying to implement my request to concentrate the logic of obtaining a type array into `TypeOfNode` in 86bfb986d185b4a01104753a20c9987591d855f3, but this cannot work - the `Object` here means: [any valid interop value](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isValidValue(java.lang.Object)) and Java arrays aren't such values.

Such a code change isn't likely to pass the **Truffle** `assert` checks as for example this [failure indicates](https://github.com/enso-org/enso/actions/runs/11942161547/job/33288566047?pr=11583#step:7:3194)

The change has to be done differently. I'll prepare a sketch of a patch.

Reply:
I can wrap the result in `EnsoObject`

Reply:
> I can wrap the result in EnsoObject

Interesting idea, yes that would fix the **Truffle assertion errors**:
- use `ArrayLikeHelpers.wrapEnsoObjects` 
- let's see what other failures we get

Reply:
I guess, I should wait for #11618

Reply:
- yes, use #11618 API now when it has been integrated